bindings/go/oci/ctf/store.go

```diff
@@ -0,0 +1,269 @@
+package oci
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"slices"
+
+	"github.com/opencontainers/go-digest"
+	ociImageSpecV1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/errdef"
+
+	"ocm.software/open-component-model/bindings/go/blob"
+	"ocm.software/open-component-model/bindings/go/ctf"
+	v1 "ocm.software/open-component-model/bindings/go/ctf/index/v1"
+	ociblob "ocm.software/open-component-model/bindings/go/oci/blob"
+	"ocm.software/open-component-model/bindings/go/oci/internal/ocireference"
+	"ocm.software/open-component-model/bindings/go/oci/spec"
+)
+
+// NewFromCTF creates a new Store instance that wraps a CTF (Common Transport Format) archive.
+// This ctf.CTF archive acts as an OCI repository interface for component versions stored in the CTF.
+func NewFromCTF(store ctf.CTF) *Store {
+	return &Store{archive: store}
+}
+
+// Store implements an OCI Store interface backed by a CTF (Common Transport Format).
+// It provides functionality to:
+// - Resolve and Tag component version references using the CTF's index archive
+// - Handle blob operations (Fetch, Exists, Push) through the CTF's blob archive
+// - Emulate an OCM OCI Repository for accessing component versions stored in the CTF
+type Store struct {
+	archive ctf.CTF
+}
+
+// TargetResourceReference returns the source reference as-is, as this archive doesn't modify references.
+func (s *Store) TargetResourceReference(srcReference string) (string, error) {
+	return srcReference, nil
+}
```

Comment:
I expected this function to be some implementation for an interface, but I cannot find the interface^^ Why does this function exist?